server/task_executor.go

```diff
@@ -166,10 +166,18 @@ func executeMigration(ctx context.Context, server *Server, task *api.Task, state
 	}
 
 	if task.Type == api.TaskDatabaseDataUpdate && task.Instance.Engine == db.MySQL {
-		_, err := setMigrationIDAndEndBinlogCoordinate(ctx, driver, task, server.store, migrationID)
+		updatedTask, err := setMigrationIDAndEndBinlogCoordinate(ctx, driver, task, server.store, migrationID)
 		if err != nil {
 			return 0, "", errors.Wrap(err, "failed to update the task payload for MySQL rollback SQL")
 		}
+		server.RollbackRunner.generateMap.Store(updatedTask.ID, updatedTask)
+		// Do not block on sending the signal if the channel is full.
+		// In that case, the runner is slower and will be signaled anyway.
+		select {
```

Comment:
Per comment above, we can get rid of this code block.

Reply:
It will block. consider that the runner generates 1 rollback SQL per second, and there are 3 DML tasks per second.

The first is being handled by the runner, the second is buffered in the channel, and the third is blocked.